### PR TITLE
feat(MarkerClusterer): Support for MarkerClusterPlus API (deprecated)

### PR DESCRIPTION
**Pull request by farrrr**
http://google-maps-utility-library-v3.googlecode.com/svn/trunk/markerclustererplus/docs/reference.html

### CODE REVIEW

File: src/creators/MarkerClustererCreator.js [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+})
+export default class MarkerClustererCreator extends Component {
+
+  static propTypes = {
+    mapHolderRef: PropTypes.instanceOf(GoogleMapHolder).isRequired,
+    markerClusterer: PropTypes.object.isRequired,
+  }
+
+  static _createMarkerClusterer (mapHolderRef, markerClustererProps) {
+    const markerClusterer = new MarkerClusterer(mapHolderRef.getMap(), [], composeOptions(markerClustererProps, [
```
Comment (tomchentw) [inline src/creators/MarkerClustererCreator.js]: Do we have a similar GitHub hosted open sourced repo for `MarkerClusterer` just like `InfoBox` does?

File: src/creators/MarkerCreator.js [hunk 2]
```diff
@@ -89,7 +90,13 @@ export default class MarkerCreator extends Component {
       "zIndex",
     ]));
 
-    marker.setMap(mapHolderRef.getMap());
+    if (anchorHolderRef) {
+      if ('MarkerClusterer' === anchorHolderRef.getAnchorType()) {
+        anchorHolderRef.getAnchor().addMarker(marker);
```
Comment (tomchentw) [inline src/creators/MarkerCreator.js]: Wait. What's the point of adding marker here?

Comment (tomchentw) [inline src/creators/MarkerCreator.js]: Sorry I got it LOL

File: src/creators/MarkerClustererCreator.js [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+  enableRetinaIcons   (enableRetinaIcons, component) { component.getMarkerClusterer().setEnableRetinaIcons(enableRetinaIcons); },
+  gridSize            (gridSize, component) { component.getMarkerClusterer().setGridSize(gridSize); },
+  ignoreHidden        (ignoreHidden, component) { component.getMarkerClusterer().setIgnoreHidden(ignoreHidden); },
+  imageExtension      (imageExtension, component) { component.getMarkerClusterer().setImageExtension(imageExtension); },
+  imagePath           (imagePath, component) { component.getMarkerClusterer().setImagePath(imagePath); },
+  maxZoom             (maxZoom, component) { component.getMarkerClusterer().setMaxZoom(maxZoom); },
+  minimumClusterSize  (minimumClusterSize, component) { component.getMarkerClusterer().setMinimumClusterSize(minimumClusterSize); },
+  styles              (styles, component) { component.getMarkerClusterer().setStyles(styles); },
+  title               (title, component) { component.getMarkerClusterer().setTitle(title); },
+  zoomOnClick         (zoomOnClick, component) { component.getMarkerClusterer().setZoomOnClick(zoomOnClick); },
```
Comment (tomchentw) [inline src/creators/MarkerClustererCreator.js]: Provide `options` here? No?

**Comment by tomchentw** [comment]
It looks great in general. Great job @farrrr !!

File: src/creators/MarkerClustererCreator.js (hunk 1 above)
Comment (farrrr) [inline src/creators/MarkerClustererCreator.js]: I think it's a good idea. But where is InfoBox repo?

File: src/creators/MarkerClustererCreator.js (hunk 3 above)
Comment (farrrr) [inline src/creators/MarkerClustererCreator.js]: Sorry, I don't know what u mean?

File: src/creators/MarkerClustererCreator.js (hunk 1 above)
Comment (tomchentw) [inline src/creators/MarkerClustererCreator.js]: Source at `InfoBoxCreator`: http://git.io/vCWHS

npm of the module: https://www.npmjs.com/package/google-maps-infobox

File: src/creators/MarkerClustererCreator.js (hunk 3 above)
Comment (tomchentw) [inline src/creators/MarkerClustererCreator.js]: Every official google maps instance provide a public API call `setOptions`: http://git.io/vCWHh. Does marker clusterer have the same API? If yes, please add it as others. If no, just leave it as is now. :+1:

File: src/creators/MarkerClustererCreator.js (hunk 1 above)
Comment (farrrr) [inline src/creators/MarkerClustererCreator.js]: ok, I try to make it. But maybe next week. it's too busy now :)

Comment (tomchentw) [inline src/creators/MarkerClustererCreator.js]: haha sure

**Comment by RoMay** [comment]
Great, the cluster is exactly what I am looking for! 
@tomchentw is there still something to do? Or is the PR ready for merge?

**Comment by tomchentw** [comment]
@RoMay would you like to help out with this comment? https://github.com/tomchentw/react-google-maps/pull/124#discussion_r41710818

**Comment by mbahoshy** [comment]
@tomchentw I've been watching this pr, any chance of getting it merged soon ?

**Comment by farrrr** [comment]
ok... I start to rewrite it now...sorry for late.

**Comment by taiga418** [comment]
This would be a nice feature to have. thanks!!

File: src/creators/MarkerClustererCreator.js (hunk 3 above)
Comment (farrrr) [inline src/creators/MarkerClustererCreator.js]: MarkerClusterPlus doesn't provide `setOptions` API..

**Comment by farrrr** [comment]
Rewrite this feature with #146 

provide more APIs. and move MarkerClusterPlus to addons.

so close it.

**Comment by tomchentw** [comment]
Release [v4.4.0](https://github.com/tomchentw/react-google-maps/releases/tag/v4.4.0) in #146 

screen shot 2015-11-19 at 10 58 43 am

**Pull request closed without merging**